notebooks/WFC3/point_spread_function/mast_api_psf.py

```diff
@@ -219,45 +303,100 @@ def make_dataURIs(obs, detector, file_suffix, sizes={'unsat': 51, 'sat': 101}):
 
     # Loop through obs to make dataURIs
     dataURIs = []
-    pixel_offset = 1 # centers sources
-    mask_full_frame = (obs['subarray'] == 0).data # only support full frame
-    for row in obs[mask_full_frame]:
+    for row in tqdm.tqdm(obs, total=len(obs)):
         # Unpack values
         iden = row['id']
         root = row['rootname']
         if detector == 'WFPC2':
             filt = row['filter_1']
         else:
             filt = row['filter']
-        x = row['psf_x_center'] - pixel_offset
-        y = row['psf_y_center'] - pixel_offset
         chip = row['chip']
         qfit = row['qfit']
         if qfit > 0:
-            size = sizes['unsat']
+            size = unsat_size
         else:
-            size = sizes['sat']
-            
-        # If UVIS use chip to asign correct sci ext
+            size = sat_size
+        subarray = row['subarray']
+        
+        # If UVIS use chip to assign correct fits ext
         if detector == 'UVIS':
-            if chip == '2':
-                sci_ext = 1
-            elif chip == '1':
-                sci_ext = 4
-                if y >= 2051:
-                    y -= 2051 - 3 # another offset to center UVIS1 sources
-        # Else chip is the correct sci ext
+            if chip == '1' and subarray == 0:
+                fits_ext = 4
+            else:
+                fits_ext = 1
+        # Else chip is the correct fits ext
         else:
-            sci_ext = chip
+            fits_ext = chip
             
         # Make dataURIs for each suffix
         for suffix in file_suffix:
-            file_read = f'red={root}_{suffix}[{sci_ext}]'
+            if suffix in ['raw', 'd0m']:
+                coord_suffix = 'raw'
+            else:
+                coord_suffix = 'cal'
+            x = row[f'x_{coord_suffix}']
+            y = row[f'y_{coord_suffix}']
+
+            file_read = f'red={root}_{suffix}[{fits_ext}]'
             cutout = f'size={size}&x={x}&y={y}&format=fits'
             file_save = f'{root}_{iden}_{filt}_{suffix}_cutout.fits'
             dataURI = f'{dataURI_base}?{file_read}&{cutout}/{file_save}'
-            dataURIs.append(("uri", dataURI))
+            dataURIs.append(dataURI)
     
-    n_subarray_sources = (~mask_full_frame).sum()
-    print(f'Found {n_subarray_sources} subarray sources in queried data.')
     return dataURIs
+
+
+def convert_dataURIs_to_dataURLs(dataURIs):
+    """
+    Convert dataURIs to URLs for the WFC3 and WFPC2 PSF databases' sources.
+
+    Use the archive url, the hla folder, and the imagename parameter.
+    
+    Parameters
+    ----------
+    dataURIs : list
```

Comment:
add support for a single string input as well, you can check if dataURIs is a string and put it into a list at the start of the function